lib/rules/named-grid-areas-no-invalid/index.js

```diff
@@ -0,0 +1,150 @@
+// @ts-nocheck
+
+'use strict';
+
+const _ = require('lodash');
+const declarationValueIndex = require('../../utils/declarationValueIndex');
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+const valueParser = require('postcss-value-parser');
+
+const ruleName = 'named-grid-areas-no-invalid';
+
+const messages = ruleMessages(ruleName, {
+	noEmptyRows: () => 'empty row',
```

Comment:
Let's make these:
- consistent with other rules
- consistent with the spec

Like so:

- `expectedToken` - `Expected cell token within string`
- `expectedSameNumber` - `Expected same number of cell tokens`
- `expectedNoRectangle` - `Expected single filled-in rectangle for "${name}"`

This breaks with our existing convention of using `rejected` for `*-no-*` rules, but I think we made a mistake by introducing double negatives.

Reply:
we can change the name of the rule, if there is a better name for this rule.

Reply:
Rule names follow [a two-part naming convention](https://stylelint.io/user-guide/rules/about#names). 

We've found that [focused rules work best](https://stylelint.io/user-guide/rules/about#about-rules).

We could further focus this rule on `strings` as that is what the above validations are referencing, i.e:

```json
{
  "rules": {
    "named-grid-areas-string-no-invalid": true
  }
}
```

Where `named-grid-areas-string` is the _thing_ and `no-invalid` the _check_. And where `string` and `named-grid-areas` come from [the spec](https://drafts.csswg.org/css-grid/#grid-template-areas-property).

What do you think?

Reply:
I think this is a bit more clear:
`expectedSameNumber` - `Expected same number of cell tokens in each string`

Reply:
should it be `expectedRectangle` instead of `expectedNoRectangle`?

Reply:
Is there another member of stylelint who can comment on what the name of the rule `named-grid-areas-no-invalid'` should be?

Reply:
the name of the rule is the only thing that is very difficult to change later on.

Reply:
> Is there another member of stylelint who can comment on what the name of the rule `named-grid-areas-no-invalid` should be?

Yes. Pull requests require two reviews to be merged.

> the name of the rule is the only thing that is very difficult to change in the future.

Yes, we try to get them right the first time because of this. Hence the documented conventions.